spine_items/tool/widgets/options_widgets.py

```diff
@@ -330,33 +329,29 @@ def _do_create_sysimage(self, tool):
         precompile_statements_file = self._get_precompile_statements_filepath()
         with open(loaded_modules_file, 'r') as f:
             modules = f.read()
-        code = f"""
-            using Pkg;
-            project_dir = dirname(Base.active_project());
-            cp(joinpath(project_dir, "Project.toml"), joinpath(project_dir, "Project.backup"); force=true);
-            cp(joinpath(project_dir, "Manifest.toml"), joinpath(project_dir, "Manifest.backup"); force=true);
-            try
-                modules = split("{modules}", " ");
-                Pkg.add(modules);
-                Pkg.add("PackageCompiler");
-                @eval import PackageCompiler
-                Base.invokelatest(
-                    PackageCompiler.create_sysimage,
-                    Symbol.(modules);
-                    sysimage_path="{self.sysimage_path}",
-                    precompile_statements_file="{precompile_statements_file}"
-                )
-            finally
-                cp(joinpath(project_dir, "Project.backup"), joinpath(project_dir, "Project.toml"); force=true);
-                cp(joinpath(project_dir, "Manifest.backup"), joinpath(project_dir, "Manifest.toml"); force=true);
-            end
-        """
+        code = f"""using Pkg;
+project_dir = dirname(Base.active_project());
+cp(joinpath(project_dir, "Project.toml"), joinpath(project_dir, "Project.backup"); force=true);
+cp(joinpath(project_dir, "Manifest.toml"), joinpath(project_dir, "Manifest.backup"); force=true);
+try
+    modules = split("{modules}", " ");
+    Pkg.add(modules);
+    Pkg.add("PackageCompiler");
+    @eval import PackageCompiler
+    Base.invokelatest(
+        PackageCompiler.create_sysimage,
+        Symbol.(modules);
```

Comment:
So in the newest PackageCompiler we can skip the first argument to `create_sysimage()` (that is, the list of modules to include in the image) and in that case it loads the modules from the current active project (https://julialang.github.io/PackageCompiler.jl/stable/sysimages.html#Creating-a-sysimage-using-PackageCompiler). I am unsure if skipping the argument would work better than giving it explicitely as we do, but maybe we can consider it in the future. It would require a lot of testing and experimentation to be sure, so imo better to keep the current code for now...

Reply:
Alright, we will leave it as is then.